models.py

```diff
@@ -286,16 +293,87 @@ class GalleyProofing(models.Model):
                   'can give the proofreader to complete their task you can '
                   'add it here.',
     )
-    proofed_files = models.ManyToManyField('core.Galley')
-    notes = models.TextField()
-    annotated_files = models.ManyToManyField('core.File')
+    proofed_files = models.ManyToManyField('core.Galley', blank=True)
+    notes = models.TextField(blank=True)
+    annotated_files = models.ManyToManyField('core.File', blank=True)
+    
+    class Meta:
+        ordering = ('assigned', 'accepted', 'pk')
 
     def __str__(self):
         return 'Proofing for Article {0} by {1}'.format(
             self.round.article.title,
             self.proofreader.full_name(),
         )
 
+    def cancel(self):
+        self.cancelled = True
+        self.completed = timezone.now()
+        self.save()
+
+    def reset(self):
+        self.cancelled = False
+        self.completed = None
+        self.accepted = None
+        self.save()
+
+    def complete(self):
+        self.completed = timezone.now()
+        self.accepted = timezone.now()
+        self.save()
+
+    def unproofed_galleys(self, galleys):
+        check = []
+        proofed_files = self.proofed_files.all()
+
+        for galley in galleys:
+            if galley not in proofed_files:
+                check.append(galley)
+
+        return check
+
+
+    @property
+    def time_to_due(self):
+        due = self.due - timezone.now()
+
+        if due.days == 0:
+            return 'Due Today'
+        
+        if due < timedelta(0):
+            return 'Overdue'
+
+        return '{} days'.format(due.days)
+
+    FRIENDLY_STATUSES = {
+        "assigned": "Awaiting response from the proofreader.",
+        "accepted": "Proofreader has accepted task, awaiting completion.",
+        "declined": "Proofreader has declined this task.",
+        "completed": "The proofreader has completed their task. "
+                     "You should review their response.",
+        "unknown": "Task status unknown.",
+        "cancelled": "Task was cancelled by manager/editor."
+    }
+
+    @property
+    def status(self):
+        if self.cancelled:
+            return 'cancelled'
+        elif self.assigned and self.accepted and not self.completed:
+            return 'accepted'
+        elif self.assigned and self.completed and not self.accepted:
+            return 'declined'
+        elif self.completed:
```

Comment:
since we have made the other branches not interdependent we might as well do that here (`self.assigned and self.completed and self.accepted`)

Reply:
Done.